### PR TITLE
add nightly build github action

### PR DESCRIPTION
**Pull request by edward-io**
tested on fork, produces https://pypi.org/project/torchsnapshot-nightly

inspired by torchrec: https://github.com/pytorch/torchrec/blob/main/.github/workflows/nightly_build_cpu.yml

### CODE REVIEW

File: .github/workflows/nightly_build_cpu.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      - name: Upload to PyPI
+        shell: bash -l {0}
+        env:
+          PYPI_USER: ${{ secrets.PYPI_USER }}
+          PYPI_TOKEN: ${{ secrets.PYPI_TOKEN }}
+        run: |
+          set -eux
+          conda activate test
+          pip install twine
+          python setup.py install --nightly sdist bdist_wheel
```
Comment (colin2328) [inline .github/workflows/nightly_build_cpu.yaml:72]: do you need both sdist and bdist wheel?

File: .github/workflows/nightly_build_cpu.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          conda install pytorch cpuonly -c pytorch-nightly
+          pip install -r requirements.txt
+          pip install -r dev-requirements.txt
+          pip install -e ".[dev]"
+      - name: Run unit tests
+        shell: bash -l {0}
+        run: |
+          set -eux
+          conda activate test
+          pytest tests -vv
```
Comment (colin2328) [inline .github/workflows/nightly_build_cpu.yaml:40]: do you not need to install torchsnapshot before testing? don't you want to test the pip installed code?

Comment (edward-io) [reply .github/workflows/nightly_build_cpu.yaml:40]: installation happens on line 34

File: torchsnapshot/__init__.py [hunk 3]
```diff
@@ -13,7 +13,7 @@
 from .stateful import Stateful
 
 
-__version__ = "0.0.2"
+__version__ = "0.0.3"
```
Comment (ananthsub) [inline torchsnapshot/__init__.py:16]: please rebase after https://github.com/facebookresearch/torchsnapshot/pull/8/

**Pull request closed without merging**